probe-rs/src/architecture/riscv/mod.rs

```diff
@@ -670,39 +688,39 @@ impl MemoryInterface for Riscv32<'_> {
     }
 
     fn write_word_64(&mut self, address: u64, data: u64) -> Result<(), Error> {
-        self.interface.write_word_64(address, data)
+        self.halted_access(|core| core.interface.write_word_64(address, data))
```

Comment:
Is this a necessity for all cores? This will absolutely trash performance :/

Reply:
no idea

Reply:
actually I think this might just be necessary when we need the progbuf

Reply:
Yeah so this made the PR grow quite a bit, but `hide whitespaces` helpfully removes 80% of the changes.